devtools/x-core/src/errors.rs

```diff
@@ -34,6 +39,13 @@ impl SystemError {
         }
     }
 
+    pub fn from_hex(context: impl Into<Cow<'static, str>>, err: FromHexError) -> Self {
```

Comment:
Just wondering why this was needed?

Reply:
Hmm, not sure I totally understand your question -- we end up parsing the Git hash that's presented, and it's possible to have errors during that process...

Reply:
Got it, new error type that exist if the hash results in an error in interacting with git.   Is it more than just the hash?  Perhaps the hash is well formed but doesn't exist in the git tree.   In that case is it a git error and not a hash error?